frontend/src/shared/util/dataset-utils.ts

```diff
@@ -16,7 +16,7 @@
 
 import { DatasetType, IDataset } from '../model';
 
-export type DatasetContext = IDataset & Required<Pick<IDataset, 'name' | 'type'>>;
+export type DatasetContext = IDataset & Required<Pick<IDataset, 'name' | 'type' | 'scope'>>;
```

Comment:
See my comment above about `scope` for this function. A pox on your linter's family for not catching this but this is invalid I'm pretty sure. It's fine to say the result will have a field `scope` but it needs to be a `string` and this function is potentially setting `undefined` by saying `matches?.scope`.